Source/Lib/Codec/EbEncHandle.c

```diff
@@ -625,33 +627,42 @@ EB_ERRORTYPE EbSetThreadManagementParameters(
     processorGroup lpgroup[16];
     memset(lpgroup, 0, 16* sizeof(processorGroup));
 
-    FILE *fin = fopen("/proc/cpuinfo", "rt");
-    if (fin) {
-        int processor_id = 0, socket_id = 0;
-        while (!feof(fin)) {
-            char line[128];
-            if (!fgets(line, sizeof(line), fin)) break;
-            if(strncmp(line, PROCESSORID, processor_id_len) == 0) {
-                char* p = line + processor_id_len;
-                while(*p < '0' || *p > '9') p++;
-                processor_id = strtol(p, NULL, 0);
-            }
-            if(strncmp(line, PHYSICALID, physical_id_len) == 0) {
-                char* p = line + physical_id_len;
-                while(*p < '0' || *p > '9') p++;
-                socket_id = strtol(p, NULL, 0);
-                if (socket_id < 0 || socket_id > 15) {
-                    fclose(fin);
-                    return EB_ErrorInsufficientResources;
+    int fd = open("/proc/cpuinfo", O_RDONLY | O_NOFOLLOW, "rt");
+    if (fd > 0) {
+        struct stat file_stat;
+        if (fstat(fd, &file_stat) != -1) {
+            if (S_ISREG(file_stat.st_mode) != 0) {
```

Comment:
these three if() can be combined to reduce nest if()

Reply:
the ifs depend on the value in the functions of the previous ifs

Reply:
struct stat file_stat; 
if(fd >=0 && fstat(fd, &file_stat) ==0 && S_ISREG(file_stat.st_mode) != 0)
with && all the functions are executed.

Reply:
I did not know that. Will promptly be fixed :)

Reply:
Done

Reply:
I realized that if(fd >= 0) needs to be separate or else it would not hit close(fd) when fstat(fd, &file_stat) !=0